readthedocs/proxito/middleware.py

```diff
@@ -59,6 +62,15 @@ def map_host_to_project_slug(request):
             request, 'core/dns-404.html', context={'host': host}, status=400
         )
 
+    if external_domain in host:
+        # Serve custom versions on external-host-domain
+        if external_domain_parts == host_parts[1:]:
+            project_slug, version_slug = host_parts[0].split('--', 1)
```

Comment:
This will probably fail with projects with `--` on their slug. It's probably and edge case, that we don't want to consider, though. Maybe 3 dashes is better?

Reply:
How our code will handle this case?

Reply:
We strip double `--`'s in slugs, so we should be fine.

Reply:
If we really want to, we could use another `.` in the domain. We just have to make sure we get the right wildcard cert (eg. a cert for `*.org.readthedocs.build` is not good for `*.*.org.readthedocs.build`).